tests/python/pants_test/backend/python/tasks2/test_python_run_integration.py

```diff
@@ -55,6 +61,70 @@ def test_get_env_var(self):
     self.assert_success(pants_run)
     self.assertEquals(var_val, pants_run.stdout_data.strip())
 
+  def test_pants_run_interpreter_selection_with_pexrc(self):
+    py27 = '2.7'
+    py3 = '3'
+    if self.has_python_version(py27) and self.has_python_version(py3):
+      print('Found both python {} and python {}. Running test.'.format(py27, py3))
+      py27_path, py3_path = self.python_interpreter_path(py27), self.python_interpreter_path(py3)
+      with setup_pexrc_with_pex_python_path(os.path.join(os.path.dirname(sys.argv[0]), '.pexrc'), [py27_path, py3_path]):
+        with temporary_dir() as interpreters_cache:
+          pants_ini_config = {'python-setup': {'interpreter_cache_dir': interpreters_cache}}
+          pants_run_27 = self.run_pants(
+            command=['clean-all', 'run', '{}:main_py2'.format(os.path.join(self.testproject, 'python_3_selection_testing'))],
+            config=pants_ini_config
+          )
+          self.assert_success(pants_run_27)
+          # Interpreter selection for Python 2 is problematic in CI due to multiple virtualenvs in play.
+          if not os.getenv('CI'):
+            assert py27_path.split(py27)[0] in pants_run_27.stdout_data
+          pants_run_3 = self.run_pants(
+            command=['clean-all', 'run', '{}:main_py3'.format(os.path.join(self.testproject, 'python_3_selection_testing'))],
+            config=pants_ini_config
+          )
+          self.assert_success(pants_run_3)
+          # Protection for when the sys.executable path underlies a symlink pointing to 'python' without '3'
+          # at the end of the basename.
+          assert py3_path.split(py3)[0] in pants_run_3.stdout_data
```

Comment:
Prefer `unittest` constructs, `self.assertIn` in this case (https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertIn). Although not documented, there is consensus that using `py.test` `assert` magic mixed in with unittest (xunit) style testing is not desirable.